src/Config/Parser/AnnotationParser.php

```diff
@@ -149,13 +151,13 @@ private static function classAnnotationsToGQLConfiguration(
                             throw new InvalidArgumentException(sprintf('The annotation @Connection on class "%s" can only be used on class implementing the ConnectionInterface.', $graphClass->getName()));
                         }
 
-                        if (!($classAnnotation->edge xor $classAnnotation->node)) {
-                            throw new InvalidArgumentException(sprintf('The annotation @Connection on class "%s" is invalid. You must define the "edge" OR the "node" attribute.', $graphClass->getName()));
+                        if (!(isset($classAnnotation->edge) xor isset($classAnnotation->node))) {
+                            throw new InvalidArgumentException(sprintf('The annotation @Connection on class "%s" is invalid. You must define either the "edge" OR the "node" attribute, but not both.', $graphClass->getName()));
                         }
 
-                        $edgeType = $classAnnotation->edge;
+                        $edgeType = $classAnnotation->edge ?? false;
                         if (!$edgeType) {
-                            $edgeType = sprintf('%sEdge', $gqlName);
```

Comment:
I prefer `sprintf` above concatenation when their not really a matter of performance. `sprintf` it more readable and can give some more information like `%d` for example say that we waiting for an int value here.

Reply:
@mcg-web @Vincz 

Concatenation (or interpolation) always converts value to string, exactly as the `%s` placeholder does. 
So the `sprintf('%sEdge', $gqlName)` doesn't eventually provide more information, than concatenation.

I know 2 scenarios, when `sprintf`makes sense:

1) When using placeholders different from `%s`, for example `%d`, `%b`, `%f` etc. We rarely use `sprintf` with other placeholder types. Currently we have in this project 215 `sprintf`'s with `%s` and only 2 with `%d`. And this is after I replaced many `sprintf` calls with concatenation. 

2) When variables are too long or require extra inline calculation, e.g.:
```php
sprintf("service('%s')", self::formatNamespaceForExpression($graphClass->getName()))
```
---

In all other cases it just brings an unnecessary overhead (function call), because I dont see, how this:
```php
sprintf('Hello, my name is %s and i am %s years old. I am from %s.', $name, $age, $place);
```
is more readable, than this:
```php
"Hello, my name is $name and i am $age years old. I am from $place.";
```
or this:
```php
"Hello, my name is {$name} and i am {$age} years old. I am from {$place}.";
```

I am against using of language constructs, unless it's really necessary or it add's some readability. I think we overuse the `sprintf` function. Look at the line 470 (file `AnnotationParser.php`):
```php
throw new InvalidArgumentException(sprintf('The "resolveType()" method on class must be static and public. Or you must define a "resolveType" attribute on the @Union annotation.'));
```
Why do we have here `sprintf`? It doesn't even have placeholders.

Reply:
For `AnnotationParser.php` it should not be used here, its a mistake, we can safely removed it. It not about what you  against here, we just try to follow [Symfony Coding standard](https://symfony.com/doc/current/contributing/code/standards.html) 
> Exception and error message strings must be concatenated using sprintf;

When doing opensource we should always try to have and follow some standards, the only accepted reasons of bypassing this is when its mater of performance or security.
Changing standards can always be done but it should be in separates PRs to ease reviewer works and stay focus on PR main issue.

Reply:
Symfony Coding Standards says only 1 sentence about `sprintf`:
> Exception and error message strings must be concatenated using sprintf;

This example is neither exception nor error message:
```php
$edgeType = sprintf('%sEdge', $gqlName);
```

Reply:
@mcg-web 

> It not about what you against here

Well, you said "I prefer `sprintf`" and a said "I am against `sprintf`" 😀
There are no standarts about `sprintf`, except when using for exceptions and errors.

Reply:
"I prefer" was just a way of being polite and opening the way to discussion but next time I can be more straightforward if this can help to go faster :+1:

Reply:
I prefer straightforwardness 🤖

Reply:
:smile: :+1: